token-swap/program/src/curve/stable.rs

```diff
@@ -0,0 +1,467 @@
+        Ok(Self {
+            trade_fee_numerator: u64::from_le_bytes(*trade_fee_numerator),
+            trade_fee_denominator: u64::from_le_bytes(*trade_fee_denominator),
+            owner_trade_fee_numerator: u64::from_le_bytes(*owner_trade_fee_numerator),
+            owner_trade_fee_denominator: u64::from_le_bytes(*owner_trade_fee_denominator),
+            owner_withdraw_fee_numerator: u64::from_le_bytes(*owner_withdraw_fee_numerator),
+            owner_withdraw_fee_denominator: u64::from_le_bytes(*owner_withdraw_fee_denominator),
+            host_fee_numerator: u64::from_le_bytes(*host_fee_numerator),
+            host_fee_denominator: u64::from_le_bytes(*host_fee_denominator),
+            amp: u64::from_le_bytes(*amp),
+        })
+    }
+}
+
+impl DynPack for StableCurve {
+    fn pack_into_slice(&self, output: &mut [u8]) {
+        let output = array_mut_ref![output, 0, 72];
+        let (
+            trade_fee_numerator,
+            trade_fee_denominator,
+            owner_trade_fee_numerator,
+            owner_trade_fee_denominator,
+            owner_withdraw_fee_numerator,
+            owner_withdraw_fee_denominator,
+            host_fee_numerator,
+            host_fee_denominator,
+            amp,
+        ) = mut_array_refs![output, 8, 8, 8, 8, 8, 8, 8, 8, 8];
+        *trade_fee_numerator = self.trade_fee_numerator.to_le_bytes();
+        *trade_fee_denominator = self.trade_fee_denominator.to_le_bytes();
+        *owner_trade_fee_numerator = self.owner_trade_fee_numerator.to_le_bytes();
+        *owner_trade_fee_denominator = self.owner_trade_fee_denominator.to_le_bytes();
+        *owner_withdraw_fee_numerator = self.owner_withdraw_fee_numerator.to_le_bytes();
+        *owner_withdraw_fee_denominator = self.owner_withdraw_fee_denominator.to_le_bytes();
+        *host_fee_numerator = self.host_fee_numerator.to_le_bytes();
+        *host_fee_denominator = self.host_fee_denominator.to_le_bytes();
+        *amp = self.amp.to_le_bytes();
+    }
+}
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+    use crate::curve::calculator::INITIAL_SWAP_POOL_AMOUNT;
+
+    #[test]
+    fn initial_pool_amount() {
+        let trade_fee_numerator = 0;
+        let trade_fee_denominator = 1;
+        let owner_trade_fee_numerator = 0;
+        let owner_trade_fee_denominator = 1;
+        let owner_withdraw_fee_numerator = 0;
+        let owner_withdraw_fee_denominator = 1;
+        let host_fee_numerator = 0;
+        let host_fee_denominator = 1;
+        let amp = 1;
+        let calculator = StableCurve {
+            trade_fee_numerator,
+            trade_fee_denominator,
+            owner_trade_fee_numerator,
+            owner_trade_fee_denominator,
+            owner_withdraw_fee_numerator,
+            owner_withdraw_fee_denominator,
+            host_fee_numerator,
+            host_fee_denominator,
+            amp,
+        };
+        assert_eq!(calculator.new_pool_supply(), INITIAL_SWAP_POOL_AMOUNT);
```

Comment:
I think we'll need to customize this as well

Reply:
+1

Reply:
Do you mean add it to the pool params?

Reply:
yes eventually, doesn't need to be done in this round though